### PR TITLE
Upgrade: decidim 0.16.1

### PR DESCRIPTION
**Pull request by aitorlb**
- Add recurring redis cron job for: decidim:open_data:export
- Add upgrade task to execute upgrade notes required tasks

**[Upgrade notes](https://github.com/decidim/decidim/blob/0.16-stable/CHANGELOG.md)**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,36 +1,39 @@
...
 
-gem 'letter_opener_web'
-gem "sprockets", "~> 3.7.2"
-
 group :development, :test do
-  gem 'decidim-dev', DECIDIM_VERSION
   gem 'byebug', platform: :mri
+  gem 'decidim-dev', DECIDIM_VERSION
+  gem 'factory_bot_rails'
+  gem 'faker'
```
Comment (tramuntanal) [inline Gemfile]: Why do we need to declare this gems?

File: Gemfile [hunk 2]
```diff
@@ -1,36 +1,39 @@
...
+DECIDIM_VERSION = '0.16.1'
+
 gem 'decidim', DECIDIM_VERSION
-gem 'sidekiq', '5.2.7'
 
+gem 'letter_opener_web'
 gem 'puma'
+gem 'sidekiq', '5.2.7'
+gem 'sidekiq-cron'
+gem 'sprockets', '~> 3.7.2'
```
Comment (tramuntanal) [inline Gemfile]: Let's comment why we are forcing the version and when can we remove this please

File: Gemfile [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
-  gem 'web-console'
   gem 'listen'
   gem 'spring'
   gem 'spring-watcher-listen'
+  gem 'web-console'
 end
 
 # Windows does not include zoneinfo files, so bundle the tzinfo-data gem
-gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
+gem 'tzinfo-data', platforms: %i[mingw mswin x64_mingw jruby]
```
Comment (tramuntanal) [inline Gemfile]: remove it?

File: Gemfile (hunk 1 above)
Comment (aitorlb) [reply Gemfile]: They were already there.

Comment (tramuntanal) [reply Gemfile]: If they were already there, why are they appearing in the diff?

Comment (aitorlb) [reply Gemfile]: Because I `rubocop`ed the Gemfile and the gems were reordered.

Comment (tramuntanal) [reply Gemfile]: ok, but do we need them? can we remove them as they are already declared by decidim?

Comment (tramuntanal) [reply Gemfile]: but do we need them? let's remove them if we don't need them.

**Pull request merged**